tinygrad/tensor.py

```diff
@@ -204,7 +204,9 @@ def reshape(self, shape, *args) -> Tensor:
   def expand(self, shape, *args) -> Tensor: return mlops.Expand.apply(self, shape=tuple(x if x != -1 else s for s,x in zip(self.shape, argfix(shape, *args))))
   def permute(self, order, *args) -> Tensor: return mlops.Permute.apply(self, order=argfix(order, *args))
   def flip(self, axis, *args) -> Tensor: return mlops.Flip.apply(self, axis=[x if x >= 0 else x+len(self.shape) for x in argfix(axis, *args)])
-  def pad(self, arg:Tuple[Tuple[int, int], ...]) -> Tensor: return mlops.Pad.apply(self, arg=arg) if any(x != (0,0) for x in arg) else self
+  def pad(self, arg:Tuple[Tuple[int, int], ...]) -> Tensor: 
```

Comment:
I mean, obviously this type signature is wrong now, right?

What if only some are tuples?

What can you remove from conv?

Reply:
Fixed edge cases, refactored pad2d and conv.